connect/runtime/src/test/java/org/apache/kafka/connect/runtime/AbstractHerderTest.java

```diff
@@ -901,6 +902,48 @@ public void testGenerateResultWithConfigValuesWithNoConfigKeysAndWithSomeErrors(
         assertInfoValue(infos, "config.extra2", "value.extra2", "error extra2");
     }
 
+    @Test
+    public void testConnectorPluginConfig() throws Exception {
+        AbstractHerder herder = partialMockBuilder(AbstractHerder.class)
+                .withConstructor(
+                        Worker.class,
+                        String.class,
+                        String.class,
+                        StatusBackingStore.class,
+                        ConfigBackingStore.class,
+                        ConnectorClientConfigOverridePolicy.class
+                )
+                .withArgs(worker, workerId, kafkaClusterId, statusStore, configStore, noneConnectorClientConfigOverridePolicy)
+                .addMockedMethod("generation")
+                .createMock();
+
+        EasyMock.expect(plugins.newConnector(EasyMock.anyString())).andReturn(new SampleSourceConnector()).anyTimes();
+        EasyMock.expect(plugins.newConverter(EasyMock.anyString())).andReturn(new SampleConverterWithHeaders()).anyTimes();
+        EasyMock.expect(plugins.newHeaderConverter(EasyMock.anyString())).andReturn(new SampleHeaderConverter()).anyTimes();
+        EasyMock.expect(plugins.newPredicate(EasyMock.anyString())).andReturn(new SamplePredicate()).anyTimes();
```

Comment:
If we want to keep the type-safe signature for `Plugins::newPredicate`:
```suggestion
        EasyMock.expect(plugins.<SourceRecord>newPredicate(EasyMock.anyString())).andReturn(new SamplePredicate()).anyTimes();
```

Note that this also requires adding an import for `org.apache.kafka.connect.source.SourceRecord` to this test class.